src/app/modules/spcp/spcp.util.ts

```diff
@@ -317,14 +318,31 @@ export const getRedirectTarget = (
  * @param encodedQuery
  * @returns
  */
-export const getRedirectTargetSpOidc = (
+export const getRedirectTargetSpcpOidc = (
   formId: string,
+  authType: FormAuthType.SP | FormAuthType.CP,
   isPersistentLogin?: boolean,
   encodedQuery?: string,
 ): RedirectTargetSpOidc => {
   // Need to cast to boolean because undefined is allowed as a valid value
-  const persistentLogin = !!isPersistentLogin
+  const persistentLogin =
+    authType === FormAuthType.SP ? !!isPersistentLogin : false
   return encodedQuery
     ? `/${formId}-${persistentLogin}-${encodedQuery}`
     : `/${formId}-${persistentLogin}`
 }
+
+// Typeguards
+
+export const isExtractedCorppassNDIPayload = (
+  payload: unknown,
+): payload is ExtractedCorppassNDIPayload => {
+  return (
+    typeof payload === 'object' &&
+    !!payload &&
+    hasProp(payload, 'userInfo') &&
+    hasProp(payload, 'userName') &&
```

Comment:
These 2 lines could be removed, the type checks below are specific enough.

Reply:
Alamak, typescript doesn't let you access an undefined property 😅 . Ignore this comment.